src/main/kotlin/com/exactpro/th2/codec/xml/XmlUtils.kt

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2021-2021 Exactpro (Exactpro Systems Limited)
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.codec.xml
+
+import com.exactpro.sf.common.messages.structures.IDictionaryStructure
+import com.exactpro.sf.common.messages.structures.IFieldStructure
+import com.exactpro.sf.common.messages.structures.IMessageStructure
+import org.w3c.dom.Document
+import org.w3c.dom.Node
+import org.w3c.dom.NodeList
+import javax.xml.xpath.XPath
+import javax.xml.xpath.XPathConstants
+
+fun NodeList.forEach(func: (Node) -> Unit) {
+    for (i in 0 until length) {
+        func(item(i))
+    }
+}
+
+fun NodeList.toList(): List<Node> {
+    val list = ArrayList<Node>(this.length)
+
+    this.forEach {
+        list.add(it)
+    }
+    return list
+}
+
+fun Node.getText(): String =
+    nodeValue ?: childNodes.let {
+        if (it.length == 0) "" else it.toList().find { node -> node.nodeType == Node.TEXT_NODE }?.nodeValue ?: ""
+    }
+
+fun Node.setText(text: String, document: Document) {
+    appendChild(document.createTextNode(text))
+}
+
+fun Node.addNode(name: String, document: Document) : Node = appendChild(document.createElement(name))
+
+fun IDictionaryStructure.getCharsetName(): String? = attributes[XmlPipelineCodec.XML_CHARSET_ATTRIBUTE]?.value
+
+fun IFieldStructure.getXmlName(): String = attributes[XmlPipelineCodec.XML_TAG_NAME_ATTRIBUTE]?.value ?: name
+fun IFieldStructure.getXPathExpression(): String? = attributes[XmlPipelineCodec.XML_X_PATH_EXPRESSION_ATTRIBUTE]?.value
+fun IFieldStructure.getAttrName(): String? = attributes[XmlPipelineCodec.XML_ATTRIBUTE_NAME_ATTRIBUTE]?.value
```

Comment:
```suggestion
fun IFieldStructure.getXmlAttrName(): String? = attributes[XmlPipelineCodec.XML_ATTRIBUTE_NAME_ATTRIBUTE]?.value
```
Use prefix to simplify identification